### PR TITLE
create conda ci test env in one step

### PR DESCRIPTION
**Pull request by msarahan**
## Description

This is a port of https://github.com/rapidsai/cuml/pull/5781 and part of https://github.com/rapidsai/build-planning/issues/22

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: dependencies.yaml [hunk 1]
```diff
@@ -451,3 +457,16 @@ dependencies:
             packages:
               - cupy-cuda11x>=12.0.0
           - {matrix: null, packages: [cupy-cuda11x>=12.0.0]}
+  test_libcuspatial:
```
Comment (msarahan) [inline dependencies.yaml:460]: The naming on these things is really bad. I'm not sure what would be better. `test_libcuspatial_includes`? Suggestions would be most welcome!

File: dependencies.yaml [hunk 2]
```diff
@@ -451,3 +457,16 @@ dependencies:
...
+        packages:
+          - libcuspatial==24.6.*
+          - libcuspatial-tests==24.6.*
+  test_cuspatial:
+    common:
+      - output_types: conda
+        packages:
+          - libcuspatial==24.6.*
+          - cuspatial==24.6.*
+          - cuproj==24.6.*
```
Comment (harrism) [inline dependencies.yaml:472]:
```suggestion
          - cuproj==24.6.*

```

File: dependencies.yaml (hunk 1 above)
Comment (harrism) [reply dependencies.yaml:460]: I find dependencies.yaml extremely confusing on the best days!

File: dependencies.yaml (hunk 2 above)
Comment (msarahan) [reply dependencies.yaml:472]: PS - best to wait for this current build to finish, especially the docs part. The tests look pretty good so far.

Comment (harrism) [reply dependencies.yaml:472]: I guess we don't have a yaml linter that catches these in pre-commit?

**Review by trxcllnt — APPROVED** [review]

File: dependencies.yaml (hunk 2 above)
Comment (jakirkham) [reply dependencies.yaml:472]: Looks like we could enable [`end-of-file-fixer` with `pre-commit`]( https://github.com/pre-commit/pre-commit-hooks?tab=readme-ov-file#end-of-file-fixer )

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Michael! 🙏

**Review by raydouglass — APPROVED** [review]

**Comment by jakirkham** [comment]
There were some CI failures, which it looks like Rob recently fixed in PR: https://github.com/rapidsai/rmm/pull/1569

Restarting the failed CI jobs

Should add the failed CI jobs are unrelated to the fix that this PR is proposing (they involved wheel builds)

**Comment by trxcllnt** [comment]
Also blocked by https://github.com/rapidsai/cudf/pull/15840

**Comment by harrism** [comment]
Those are both merged. What's holding this now?

**Comment by trxcllnt** [comment]
@harrism nothing, I just re-ran the failing jobs

**Comment by jakirkham** [comment]
Was waiting for packages from those before restarting. Looks like they were uploaded ~20mins ago. So hopefully now things clear up

**Comment by jakirkham** [comment]
Thanks all! 🙏

File: dependencies.yaml (hunk 1 above)
Comment (bdice) [reply dependencies.yaml:460]: (Sorry for the late response)

I would read `test_libcuspatial` as "Test dependencies for libcuspatial."

Comment (bdice) [reply dependencies.yaml:460]: Elsewhere in RAPIDS, we use [`depends_on_cudf`](https://github.com/rapidsai/cudf/blob/54d49fcea4e7ad73df21f0dbfe99097c635b1023/dependencies.yaml#L699), specifically for packages that are shipped by this repository. That might have been a better approach here, in retrospect.

**Pull request merged**